content/doc/pipeline/steps/index.html.haml

```diff
@@ -16,19 +16,55 @@ title: "Pipeline Steps Reference"
   %a{:href => "/doc/book/pipeline/syntax"} Pipeline Syntax
   page.
 
+.input-group.col-lg-3.mb-3
+  %input#filter.form-control.rounded{:name => "search", :placeholder => "Filter Steps", :type => "search", :autofocus => "true"}
+
 %div
   %ul
     - sections_from(steps_dir).sort_by { |it| it[0].downcase }.each do |section, file, url, children|
-      %li
+      %li.search
         %a{:href => url}
           = section
         %ul
           - children.each do |section, file, sect_url, sect_children|
-            %li
+            %li.search
               %a{:href => "#{url}##{sect_url}"}
                 = section
 
 %hr/
 
 .container
 = partial ('_feedback-footer.html')
+
+:javascript
+    function slowClear() {
+      $(".search").filter(function() {
+        setTimeout(() => $(this).show(), 20);
+      });
+    }
+
+    function debounce(func) {
+      let timer;
+      return (...args) => {
+        clearTimeout(timer);
+        timer = setTimeout(() => { func.apply(this, args); }, 500);
+      };
+    }
+
+    $("#filter").keyup(debounce(() => {
+      var input = $("#filter").val().toLowerCase();
+      if($("#filter").val() === "") {
+        slowClear();
+      } else {
+        $(".search").filter(function() {
+          $(this).toggle($(this).text().toLowerCase().indexOf(input) > -1);
+        });
+      }
+    }));
+
+    $("#filter").on("search", function() {
+      console.log("yo");
```

Comment:
yo ?

Reply:
sorry, forgot to get rid of log messages for testing